src/Features/VisualBasic/Portable/ConvertAutoPropertyToFullProperty/VisualBasicConvertAutoPropertyToFullProperty.vb

```diff
@@ -19,29 +20,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ConvertAutoPropertyToFullProperty
         Public Sub New()
         End Sub
 
-        Friend Overrides Function GetProperty(token As SyntaxToken) As SyntaxNode
-            Dim containingProperty = token.Parent.FirstAncestorOrSelf(Of PropertyStatementSyntax)
-            If containingProperty Is Nothing Then
-                Return Nothing
-            End If
+        Friend Overrides Async Function GetPropertyAsync(document As Document, span As TextSpan, cancellationToken As CancellationToken) As Task(Of SyntaxNode)
+            Dim refactoringHelperService = document.GetLanguageService(Of IRefactoringHelpersService)()
+            Dim containingProperty = Await refactoringHelperService.TryGetSelectedNodeAsync(Of PropertyStatementSyntax)(document, span, cancellationToken).ConfigureAwait(False)
```

Comment:
could potentially move this method up now.

Reply:
I need to call with specific types `PropertyStatementSyntax` and `PropertyDeclarationSyntax` (in C#) as generic args. Not sure how to do that from lang-agnostic parent (would be doable with predicate, but we removed that overload).

Reply:
it's pretty simple.  We make the base type *generic* and have the derived types instantiate it with teh appropriate type args themselves.  This is a *very* common pattern we use for features when we want to share logic at the feature layer.

Reply:
Adding a Type arg to the Abstract base, clever! Didn't consider that.